<launch>
    <!-- Topic-->
    <arg name="detections_image" default="detections_image"/>
    <arg name='image_source' default='/device_0/sensor_1/Color_0/image/data'/>

    <!-- # Arg Detector -->
    <!-- ## Path File-->
    <arg name="yolo_cfg" default="yolo_model/yolo-obj.cfg"/>
    <arg name="yolo_weights" default="yolo_model/yolo-obj.weights"/>
    <arg name="obj_names" default="yolo_model/obj.names"/>

    <!-- ## Thresholds-->
    <arg name="conf_threshold" default="0.3"/>
    <arg name="nms_threshold" default="0.4"/>


    <node name="$(anon detector_mechanical_parts)" pkg="detect_mechanical_parts" type="detector_node.py" output="screen">
        <param name="detections_image_topic" type="str" value="$(arg detections_image)"/>
        <param name="image_source_topic" type="str" value="$(arg image_source)"/>

        <!-- Detector -->
        <param name="yolo_cfg" type="str" value="$(arg yolo_cfg)"/>
        <param name="yolo_weights" type="str" value="$(arg yolo_weights)"/>
        <param name="obj_names" type="str" value="$(arg obj_names)"/>
        <param name="conf_threshold" type="double" value="$(arg conf_threshold)"/>
        <param name="nms_threshold" type="double" value="$(arg nms_threshold)"/>

    </node>

    <node name="$(anon visualizer)" pkg="detect_mechanical_parts" type="visualizer.py" output="screen">
        <param name="view_image_topic" value="$(arg detections_image)"/>
    </node> 
</launch>